src/app/Home/components/HomeComponent.js

```diff
@@ -1,7 +1,52 @@
-import React from 'react';
+import React, { Component } from 'react';
+import Modal from './onBoardingModal';
 
-const HomeComponent = () => (
-  <h1 className='home__title'>Welcome to Andela Societies </h1>
-);
+export class HomeComponent extends Component {
+  constructor() {
+    super();
 
-export default HomeComponent;
+    this.state = {
+      isShowing: false,
+      shownOnce: !!localStorage.getItem('shownonce'),
+    };
+  }
+
+  componentDidMount() {
+    const { shownOnce } = this.state;
+    if (!shownOnce) {
+      this.setState({
+        isShowing: true,
+        shownOnce: true,
+      }, () => {
+        localStorage.setItem('shownonce', true);
+      });
+    }
+  }
+
+closeModalHandler = () => {
+  this.setState({
+    isShowing: false,
+  });
+}
+
+render() {
+  const { isShowing } = this.state;
+  return (
+    <div>
+      { isShowing
+        ? <div role='presentation' onClick={this.closeModalHandler} className='back-drop' />
+        : null
+      }
```

Comment:
I had mentioned this some time back. It would be better to have this computed in the render function to improve readability e.g

```
let isShowingContent = "";
if(isShowing){
isShowingContent = <div role='presentation' onClick={this.closeModalHandler} className='back-drop' />
};
```

That way we don't have to evaluate the null path within the JSX.  Makes sense?